go/vt/vtgate/evalengine/comparisons.go

```diff
@@ -126,27 +125,43 @@ func evalResultsAreDateAndNumeric(l, r EvalResult) bool {
 
 // For more details on comparison expression evaluation and type conversion:
 // 		- https://dev.mysql.com/doc/refman/8.0/en/type-conversion.html
-func executeComparison(lVal, rVal EvalResult) (int, error) {
+func nullSafeExecuteComparison(lVal, rVal EvalResult) (comp int, isNull bool, err error) {
```

Comment:
I think we should keep this method as it was before, and add another method (`nullSafeExecuteComparison`) that calls (`executeComparison`). The new method (`nullSafeExecuteComparison`) will encapsulate the logic that was previously in `ComparisonExpr.Evaluate`:
```go
if hasNullEvalResult(lVal, rVal) {
	return 0, true, nil
}
```

This will allow us to have a simpler function for comparisons.

Reply:
After adding comparisons for tuples, we need nullSafeComparisons in all the places. Since there is no place where we would call the executeComparison from we haven't extracted out yet.